codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/middleware/factories/RequestCompressionMiddlewareFactory.java

```diff
@@ -59,31 +61,31 @@ public static Middleware build(GenerationContext context) {
                 .constraint(new RangeConstraint(0, 10485760))
                 .build();
 
+        Middleware.Relative compressionRelative = Middleware.Relative.builder()
+                .before("Hearth::HTTP::Middleware::ContentMD5")
+                .optional()
+                .build();
+
         Middleware.Builder compressionBuilder = Middleware.builder()
+                .klass("Hearth::HTTP::Middleware::RequestCompression")
+                .step(MiddlewareStackStep.AFTER_BUILD)
```

Comment:
Totally optional, but if you're up to it, I think we should also improve middleware stack steps to match 1:1 with our hearth middleware steps. We chose names similar to Go but it's kind of just confusing the way it is now.